kv/local_test_cluster.go

```diff
@@ -146,5 +122,8 @@ func (ltc *LocalTestCluster) Start(t util.Tester) {
 
 // Stop stops the cluster.
 func (ltc *LocalTestCluster) Stop() {
+	if r := recover(); r != nil {
```

Comment:
I strongly prefer `Fatalf()` to `Errorf()` as a general rule. If you have a panic after an `Errorf`, the `Errorf` message is thrown away and you only see the panic. We should only use `Errorf` when we are confident that continuing past an error will not result in a panic.

Where possible, we should add a `*testing.T` to our test structures that have a stop method. Then we can do this, which I think safely avoids the shutdown hangs for both panic and `Fatalf`:

    if t.Failed() {
        return
    }
    if r := recover(); r != nil {
        panic(r)
    }